cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -202,11 +222,28 @@ func GetIntermediates() (*x509.CertPool, error) {
 	return fulcioroots.GetIntermediates()
 }
 
-func NewClient(fulcioURL string) (api.LegacyClient, error) {
+func NewClient(fulcioURL string) (fulciopb.CAClient, error) {
+	var opts []grpc.DialOption
 	fulcioServer, err := url.Parse(fulcioURL)
 	if err != nil {
 		return nil, err
 	}
-	fClient := api.NewClient(fulcioServer, api.WithUserAgent(options.UserAgent()))
-	return fClient, nil
+	host := fulcioServer.Host
+	switch fulcioServer.Scheme {
+	case "https":
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{MinVersion: tls.VersionTLS12})))
+		if fulcioServer.Port() == "" {
+			host = fmt.Sprintf("%s:443", host)
+		}
+	default:
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
TODO: add a retry policy?

Reply:
let's do this in a separate PR, because it should also apply to calling other external services too